### PR TITLE
ipld: move for dag service to block service

### PR DESCRIPTION
**Pull request by vgonkivs**

### CODE REVIEW

File: ipld/retriever.go [hunk 1]
```diff
@@ -87,10 +89,17 @@ type retrieverSession struct {
...
 	size := len(dah.RowsRoots)
-	adder := NewNmtNodeAdder(ctx, format.NewBatch(ctx, r.dag, format.MaxSizeBatchOption(batchSize(size))))
+	adder := NewNmtNodeAdder(
+		ctx,
+		format.NewBatch(
+			ctx,
+			merkledag.NewDAGService(r.bServ),
+			format.MaxSizeBatchOption(batchSize(size)),
+		),
+	)
```
Comment (Wondertan) [inline ipld/retriever.go:99]: We should add additional NodeAdder constructor from Blockservice, context and size

File: ipld/plugin/nmt.go [hunk 2]
```diff
@@ -184,7 +184,16 @@ func prependNode(newNode ipld.Node, nodes []ipld.Node) []ipld.Node {
 	return prepended
 }
 
-func NmtNodeParser(block blocks.Block) (ipld.Node, error) {
+func Get(ctx context.Context, bGetter blockservice.BlockGetter, root cid.Cid) (ipld.Node, error) {
```
Comment (Wondertan) [inline ipld/plugin/nmt.go:187]: GetNode s.t. `plugin.GetNode`

**Review by liamsi — APPROVED** [review]
I think @Wondertan should have a final say here but this LGTM.

**Review by Wondertan — APPROVED** [review]
Very nice. Thank you. 
Just a few prettifying nits.

Also, this PR does not fully solve the attached issue. It is just a first step.

File: ipld/plugin/nmt.go (hunk 2 above)
Comment (vgonkivs) [reply ipld/plugin/nmt.go]: done

File: ipld/retriever.go (hunk 1 above)
Comment (vgonkivs) [reply ipld/retriever.go]: Changed `NewNmtNodeAdder` ctor.

File: node/p2p/ipld.go [hunk 3]
```diff
@@ -4,11 +4,9 @@ import (
...
 	blockstore "github.com/ipfs/go-ipfs-blockstore"
 	exchange "github.com/ipfs/go-ipfs-exchange-interface"
-	format "github.com/ipfs/go-ipld-format"
-	"github.com/ipfs/go-merkledag"
 )
 
-// DAG constructs IPFS's DAG Service for fetching arbitrary Merkle structures.
-func DAG(bs blockstore.Blockstore, ex exchange.Interface) format.DAGService {
-	return merkledag.NewDAGService(blockservice.New(bs, ex))
+// BlockService constructs IPFS's BlockService for fetching arbitrary Merkle structures.
```
Comment (renaynay) [inline node/p2p/ipld.go:9]: Is it IPFS?

**Review by renaynay — APPROVED** [review]
lgtm with one comment

File: node/p2p/ipld.go (hunk 3 above)
Comment (vgonkivs) [reply node/p2p/ipld.go:9]: `github.com/ipfs/go-blockservice`

**Pull request merged**